domain/user/bootstrap/bootstrap.go

```diff
@@ -39,6 +39,8 @@ func AddUser(name string, access permission.AccessSpec) (user.UUID, internaldata
 			fmt.Errorf("generating bootstrap user %q uuid: %w", name, err))
 	}
 
+	state := state.NewSharedState(internaldatabase.DefaultStatementBase{})
```

Comment:
as a wart, having a 'state' variable vs a 'state' imported package.
Which also points that having the name of the package be: domain/permission/state/shared.go leads towards awkward structures when used elsewhere.
permissions.NewSharedState would probably be a lot clearer.
permissionstate.NewShared() 
shared.NewPermissionState()
state.NewSharedPermissions()

thoughts?

Reply:
Personally, I prefer `SharedState` and therefore `NewSharedState()`. We're implying that we have a state, but it's shared from the naming and it aligns with the `State` type as well. I do agree that it can look like stuttering, but in this instance, I think I can forgive that.

Reply:
He's talking about the assignment to a `state` variable, which is also an import. It could be assigned to `st` for example.